### DISCUSSION TITLE
Setting the foreground color (text) to something with more contrast

### DISCUSSION THREAD

**Discussion by rikhuijzer** [ellisonleao/gruvbox.nvim#331] [Show and tell]
I work a lot with documents containing plain old text. For these documents, I prefer as much contrast as possible because walls of text become harder to read when the contrast between the background and foreground is less. Books are also still printed in black letters on white (or gray) pages is my thinking.

Luckily, `gruvbox` is great because it can easily be adjusted! Very nice work on that.

It took me a while to figure out how to change the main text color (the `Normal` highlight group), so in the hope of saving other people some time, here it is:

```vim
call plug#begin()
  Plug 'ellisonleao/gruvbox.nvim'
end

lua << EOF
  require("gruvbox").setup({
    contrast = "hard",
    palette_overrides = {
        -- Set the foreground color to something more readable.
        -- This is the color of the text.
        light1 = "#ebe8e8"
    }
  })
EOF
colorscheme gruvbox
```

**Comment by realyukii** [comment]
I tried your example in my `init.lua` file:
```
require("gruvbox").setup({
    contrast = "hard",
    palette_overrides = {
        -- Set the foreground color to something more readable.
        -- This is the color of the text.
        light1 = "#ebe8e8"
    }
})

vim.o.background = "dark"
vim.cmd('colorscheme gruvbox')
```

but it seems the contrast between background and foreground is still hard to read for some syntax/keyword such as `final`:
![image](https://github.com/user-attachments/assets/3de295f7-9792-4934-8669-262aa6862731)

what do I need to adjust to fix this? thanks for the example!

**Reply by realyukii** [reply]
Nvm, after I install the java lsp (jdtls), the syntax highlighting feature just work as expected (including the color)
